sway-core/src/semantic_analysis/ast_node/expression/intrinsic_function.rs

```diff
@@ -331,6 +332,56 @@ fn type_check_check_str_type(
     ))
 }
 
+fn type_check_to_str_array(
+    handler: &Handler,
+    mut ctx: TypeCheckContext,
+    kind: sway_ast::Intrinsic,
+    arguments: Vec<Expression>,
+    span: Span,
+) -> Result<(ty::TyIntrinsicFunctionKind, TypeId), ErrorEmitted> {
+    let type_engine = ctx.engines.te();
+    let engines = ctx.engines();
+
+    if arguments.len() != 1 {
+        return Err(handler.emit_err(CompileError::IntrinsicIncorrectNumArgs {
+            name: kind.to_string(),
+            expected: 1,
+            span,
+        }));
+    }
+
+    let mut ctx = ctx
+        .by_ref()
+        .with_type_annotation(type_engine.insert(engines, TypeInfo::Unknown));
+
+    let arg = arguments[0].clone();
+    let arg = ty::TyExpression::type_check(handler, ctx.by_ref(), arg)?;
+
+    let new_type = match type_engine.get(arg.return_type) {
+        TypeInfo::StringSlice => {
+            let span = arg.span.clone();
+            TypeInfo::StringArray(Length::new(span.as_str().len() - 2, span))
```

Comment:
Is `__to_str_array` supposed to be a compile-time only thing? Could there be a need for it in the future to work on any string slice, not just string literals?

It would be nice to have this new intrinsic documented, otherwise it's unclear what we are reviewing against.

Reply:
> Is __to_str_array supposed to be a compile-time only thing? 

Yes. I improved the PR description and will improve doc to make this clear.

Reply:
Done.